### PR TITLE
Issue 267

### LINKED ISSUE #267: Register with Poll Buddy page functionality
**Please describe what has to be done**
Should submit a request to the backend for creating a new account. Should also validate password requirements (TBD). Once successful, log in automatically and redirect to /groups.

### PR DESCRIPTION
**Pull request by aidan-lane**
### Description of change(s), including why:

Completed issue 267: Completed registration functionality. The user can now register for an account through Poll Buddy. Added validation for:
**username** - Between 3 and 32 characters, no special characters
**email** - must be in format ###@###.###
**password** - Greater than 6 characters, at least 1 uppercase and 1 digit.
*These validations can be changed, I just went with the most simple for now.

Also updated the backend to make the email field unique so no two users can have the same email. If there is an error (i.e. incorrect format, email already exists) then no form will be submitted.

### Type of change(s)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Any work to still do or other concerns:

- The user is logged in after registering, but some things should be updated such as the sign in button.

### Closing issues:

- Closes #267

### CODE REVIEW

**Comment by hbh7** [comment]
I've made an issue to discuss the password requirements but those are definitely good starting points. https://github.com/PollBuddy/PollBuddy/issues/298

File: PollBuddy-Server/frontend/src/pages/RegisterWithPollBuddy/RegisterWithPollBuddy.js [hunk 1]
```diff
@@ -19,18 +77,40 @@ export default class RegisterWithPollBuddy extends Component {
...
             <label htmlFor="passwordText">Password:</label>
-            <input placeholder="●●●●●●●●●●●●" className="form-control textBox" id="passwordText"/>
+            <input type="password" placeholder="●●●●●●●●●●●●" className="form-control textBox" id="passwordText"
+              onChange= {(evt) => { this.setState({password: evt.target.value}); }}/>
+            {!this.state.passValid && 
+              <div className="error">
+                <p>Invalid password. Must contain:</p>
+                <p>6 or more characters</p>
+                <p>At least 1 uppercase letter</p>
+                <p>At least 1 number</p>
```
Comment (hbh7) [inline PollBuddy-Server/frontend/src/pages/RegisterWithPollBuddy/RegisterWithPollBuddy.js:105]: Could be good as a bulleted list perhaps, not sure. Problem for later when the requirements are figured out though.

**Review by hbh7 — APPROVED** [review]
Very nice! Quite impressive seeing this working

And again I'll take a look before merging at the linting to make sure it's behaving like desired.

**Review by annieshi6 — APPROVED** [review]

**Comment by hbh7** [comment]
Yep, the lint rules needed some adjustments. One-line statements can be fine so the rules have been modified to allow that.

**Review by hbh7 — APPROVED** [review]
Looks great!

**Review by annieshi6 — APPROVED** [review]

**Pull request merged**